### PR TITLE
Add WooPay marketing opt-in for selected countries

### LINKED ISSUE #5400: Show WooPay marketing opt-in checkbox for countries that require it
### Description

Some countries requires user marketing consent before signing up to a service, while others allow us to send marketing emails just by accepting the terms of service.

Currently WooPay is available only for US users, that allows it to be included on the terms of service, so there's no problem, but we need to add a checkbox for marketing consent on WooPay sign up form when the user select one of these countries for international launch.

There's a list [here](https://github.com/threeheartsdigital/email-marketing-regulations) of which countries that currently we need this user consent before signing up, it should be only the ones that require explicit consent, this should also include the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries).

### Acceptance criteria

* When the user selects a country that requires marketing opt-in consent and choose to sign up to WooPay, show a checkbox to opt-in/opt-out from marketing e-mails.

### Designs

Something like that:

screenshot-2022-12-12-at-9 00 52-pm-1

### PR DESCRIPTION
**Pull request by alefesouza**
Fixes #5400

#### Changes proposed in this Pull Request

Add a new `Opt-in for marketing emails` (this text is not final) checkbox on WooPay sign up form when the user selects a billing address country that requires single opt-in, this includes the GDPR countries and countries that require explicit marketing consent (China, Singapore).

#### Testing instructions

This PR needs to be tested alongside 2878-gh-Automattic/woocommerce-payments-server and 1567-gh-Automattic/woopay.

It should be tested with classic checkout and blocks checkout.

**Disabling marketing**

* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) or China/Singapore.
* Click `Save my information for faster checkouts`.
* There should be a new `Opt-in for marketing emails` checkbox checked by default.
* Uncheck the `Opt-in for marketing emails` checkbox.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be unchecked.

**Enabling marketing**

* Sign out on WooPay.
* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) or China/Singapore.
* Click `Save my information for faster checkouts`.
* Let the `Opt-in for marketing emails` checkbox checked.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be checked.

**Other countries**

* Sign out on WooPay.
* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Do not select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) nor China/Singapore.
* Click `Save my information for faster checkouts`.
* There should not be a `Opt-in for marketing emails` checkbox.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be checked.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/platform-checkout-user/class-platform-checkout-save-user.php [hunk 1]
```diff
@@ -88,7 +91,10 @@ public function maybe_add_userdata_to_metadata( $metadata, $order ) {
 			$metadata['platform_checkout_secondary_last_name']  = wc_clean( $order->get_shipping_last_name() );
 			$metadata['platform_checkout_secondary_phone']      = wc_clean( $order->get_shipping_phone() );
 			$metadata['platform_checkout_secondary_company']    = wc_clean( $order->get_shipping_company() );
+			$metadata['platform_checkout_source_url']           = $platform_checkout_marketing_source_url;
```
Comment (malithsen) [inline includes/platform-checkout-user/class-platform-checkout-save-user.php:96]: I don't see platform_checkout_source_url and platform_checkout_is_blocks being used anywhere. Are we recording them for tracking purposes?

**Review by malithsen — COMMENTED** [review]
Thanks for working on this @alefesouza. Changes are functional and the code looks good to me. I had a question on how we are planning to use 2 of the meta fields we are recording. Some of the required tests are failing. Let's address them.

File: includes/platform-checkout-user/class-platform-checkout-save-user.php (hunk 1 above)
Comment (alefesouza) [reply includes/platform-checkout-user/class-platform-checkout-save-user.php:96]: It's used on 1568-gh-Automattic/woopay, I've included it on this PR so we don't need to open another one just to include these two fields.

Comment (malithsen) [reply includes/platform-checkout-user/class-platform-checkout-save-user.php:96]: ah thanks Alefe. If we are confident that we'd be collecting these two meta fields even if the proposed architecture in that PR ends up changing, I am okay with including them in this PR.

**Review by malithsen — APPROVED** [review]

**Comment by alefesouza** [comment]
Following pdpOw2-24R-p2#comment-2325, for beta we will not have the marketing opt-in checkbox as WooPay will work only for US customers, this PR may be useful in the future for the international launch.

**Pull request closed without merging**